carmajava/signal_plugin/src/main/java/gov/dot/fhwa/saxton/carma/signal_plugin/ead/trajectorytree/CoarsePathNeighbors.java

```diff
@@ -152,6 +152,12 @@ public void initialize(List<IntersectionData> intersections, int numIntersection
                     timeAtNext += extraTime;
                     if (speedAtNext < operSpeed_) {
                         speedAtNext = 2.0*(distToNext - lagDist)/(timeAtNext - (curTime + lagTime_)) - curSpeed;
+                    } else {
+                        if(intermediateNode != null) {
+                            // update intermediate node if we shift the goal node time
+                            intermediateNode = new Node(intermediateNode.getDistanceAsDouble() + extraTime * deltaSpeed,
+                                                        intermediateNode.getTimeAsDouble() + 2 * extraTime, operSpeed_);
```

Comment:
Can you explain these calculations?

Reply:
Just updated the math based on the whiteboard discussion.